hyperspy/_signals/wave_image.py

```diff
@@ -0,0 +1,165 @@
+        phase = self._deepcopy_with_new_data(np.angle(self.data))
+        phase.set_signal_type('')  # Result is a normal Image!
+        return phase
+
+    @phase.setter
+    def phase(self, phase):
+        if isinstance(phase, Image):
+            phase = phase.data
+        self.data = self.amplitude.data * np.exp(1j * phase)
+
+    @property
+    def amplitude(self):
+        """Get/set the amplitude of the data. Returns an :class:`~hyperspy._signals.Image`."""
+        amplitude = self._deepcopy_with_new_data(np.abs(self.data))
+        amplitude.set_signal_type('')  # Result is a normal Image!
+        return amplitude
+
+    @amplitude.setter
+    def amplitude(self, amplitude):
+        if isinstance(amplitude, Image):
+            amplitude = amplitude.data
+        self.data = amplitude * np.exp(1j * self.phase.data)
+
+    @property
+    def real(self):
+        """Get/set the real part of the data. Returns an :class:`~hyperspy._signals.Image`."""
+        real = self._deepcopy_with_new_data(np.real(self.data))
+        real.set_signal_type('')  # Result is a normal Image!
+        return real
+
+    @real.setter
+    def real(self, real):
+        if isinstance(real, Image):
+            real = real.data
+        self.data = real + 1j * self.imag.data
+
+    @property
+    def imag(self):
+        """Get/set the imaginary part of the data. Returns an :class:`~hyperspy._signals.Image`."""
+        imag = self._deepcopy_with_new_data(np.imag(self.data))
+        imag.set_signal_type('')  # Result is a normal Image!
+        return imag
+
+    @imag.setter
+    def imag(self, imag):
+        if isinstance(imag, Image):
+            imag = imag.data
+        self.data = self.real.data + 1j * imag
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # Make sure data is complex:
+        self.change_dtype(complex)
+
+    def get_unwrapped_phase(self, wrap_around=False, seed=None):
+        """Return the unwrapped phase as an :class:`~hyperspy._signals.Image`.
+
+        Parameters
+        ----------
+        wrap_around : bool or sequence of bool, optional
+            When an element of the sequence is  `True`, the unwrapping process
+            will regard the edges along the corresponding axis of the image to be
+            connected and use this connectivity to guide the phase unwrapping
+            process. If only a single boolean is given, it will apply to all axes.
+            Wrap around is not supported for 1D arrays.
+        seed : int, optional
+            Unwrapping 2D or 3D images uses random initialization. This sets the
+            seed of the PRNG to achieve deterministic behavior.
+
+        Returns
+        -------
+        phase_image: :class:`~hyperspy._signals.Image`
+            Unwrapped phase.
+
+        Notes
+        -----
+        Uses the :func:`~skimage.restoration.unwrap_phase` function from `skimage`.
+
+        """

```

Comment:
I think that it'll be preferable to lazy import skimage to make it optional instead of required.